tests/unit/core/reducers/test_abuse.js

```diff
@@ -0,0 +1,125 @@
+import abuseReducer, {
+  SEND_ADDON_ABUSE_REPORT,
+  initialState,
+  loadAddonAbuseReport,
+  sendAddonAbuseReport,
+} from 'core/reducers/abuse';
+import { fakeAddon } from 'tests/unit/amo/helpers';
+import {
+  createFakeAddonAbuseReport,
+  createStubErrorHandler,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  describe('reducer', () => {
+    it('initializes properly', () => {
+      const state = abuseReducer(initialState, { type: 'UNRELATED_ACTION' });
+      expect(state).toEqual({});
+    });
+
+    describe('sendAddonAbuseReport', () => {
+      it('dispatches a sendAddonAbuseReport', () => {
+        const addonSlug = 'slugs-are-gross';
+        const errorHandler = createStubErrorHandler('or-are-they');
+        const message = 'They really are';
+
+        const action = sendAddonAbuseReport({
+          addonSlug,
+          errorHandler,
+          message,
+        });
+
+        expect(action.type).toEqual(SEND_ADDON_ABUSE_REPORT);
+        expect(action.payload).toEqual({
+          addonSlug,
+          errorHandlerId: errorHandler.id,
+          message,
+        });
+      });
+
+      it('requires an addonSlug', () => {
+        expect(() => {
+          sendAddonAbuseReport({
+            errorHandler: createStubErrorHandler('or-are-they'),
+            message: 'hello!',
+          });
+        }).toThrow('addonSlug is required');
+      });
+
+      it('requires an errorHandler', () => {
+        expect(() => {
+          sendAddonAbuseReport({
+            addonSlug: 'funk-music',
+            message: 'hello!',
+          });
+        }).toThrow('errorHandler is required');
+      });
+
+      it('requires a message', () => {
+        expect(() => {
+          sendAddonAbuseReport({
+            addonSlug: 'i-am-sparticus',
+            errorHandler: createStubErrorHandler('or-are-they'),
+          });
+        }).toThrow('message is required');
+      });
+    });
+
+    describe('loadAddonAbuseReport', () => {
+      let addon;
+      let message;
+      let response;
+
+      beforeEach(() => {
+        addon = { ...fakeAddon, slug: 'Ego' };
+        message = 'I am Groot!';
+        response = createFakeAddonAbuseReport({ addon, message });
+      });
+
+      it('saves the abuse report response to the reducer', () => {
```

Comment:
I'd like to see some additional tests that make sure multiple reports for different add-ons (or the same add-on) preserve existing state since that code can break in a refactor.

Reply:
We shouldn't allow multiple abuse reports for the same add-on (I was thinking I'd add that test once I made the UI for it too) but I'll do it here. And yeah, multiple reports across add-ons should be tested, good call.